src/booking/containers/booking-date/booking-date.component.spec.ts

```diff
@@ -17,20 +38,100 @@ describe('BookingDateComponent', () => {
     .compileComponents();
     fixture = TestBed.createComponent(BookingDateComponent);
     component = fixture.componentInstance;
+    component.bookingProcess = {selectedDateOption: 0} as BookingProcess;
+    component.dateInterval = dateInterval;
+    component.configStart = configStart;
+    component.configEnd = configEnd;
+    component.formGroup = new FormBuilder().group({
+      startDate_day: '',
+      startDate_month: '',
+      startDate_year: '',
+      endDate_year: '',
+      endDate_month: '',
+      endDate_day: '',
+      dateOption: '',
+    });
+    component.setValidators();
     fixture.detectChanges();
   }));
 
-  beforeEach(() => {
-    fixture = TestBed.createComponent(BookingDateComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  });
-
   it('should create component and show the \"booking date\" info message banner', () => {
     expect(component).toBeDefined();
     const infoBannerElement = fixture.debugElement.nativeElement.querySelector('.govuk-caption-l');
-    expect(infoBannerElement.textContent).toContain(BookingTimePageText.CAPTION);
+    expect(infoBannerElement.textContent).toContain(BookingDatePageText.CAPTION);
     const headingElement = fixture.debugElement.nativeElement.querySelector('.govuk-fieldset__heading');
-    expect(headingElement.textContent).toContain(BookingTimePageText.TITLE);
+    expect(headingElement.textContent).toContain(BookingDatePageText.TITLE);
+  });
+
+  it('should show a validation error if form submitted with no option selected', () => {
+    component.formGroup.get(DateFormControl.BOOKING_DATE_TYPE).patchValue(null);
+    fixture.detectChanges();
+    const submitButton = fixture.debugElement.nativeElement.querySelector('button[type="submit"]');
```

Comment:
Can submitButton be defined outside of the test - would reduce the amount of code and would save you having to define it in every test

Reply:
I agree, if the button is shown with no condition it could go into a beforeEach

Reply:
it s good to define dependent objects and on beforeeach , but HTML element creating is not , in other part of the project is the same as I did, if there is a different implementation as you mentioned in the project could you please share it? the other reason is that element would lead null on fixture.